src/nvim/api/vim.c

```diff
@@ -2314,3 +2314,25 @@ void nvim_error_event(uint64_t channel_id, Integer lvl, String data)
   // if we fork nvim processes as async workers
   ELOG("async error on channel %" PRId64 ": %s", channel_id, data.size ? data.data : "");
 }
+
+/// Set info for the completion candidate index.
+/// If the index is the currently selected index,
+/// a floating window will display the info text.
```

Comment:
can expand a bit with the purpose of the return value "if the info was shown in a window, then the window and buffer ids are returned for further customization. If the text was not shown, an empty dict is returned"

Where we don't mention "floating" as this should apply regardless if the text was shown in the split preview or the float window, and a plugin needs to be ware of both possibility.